sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -1335,4 +1352,41 @@ export class BlobClient extends StorageClient {
     (response as any).blobDownloadStream = undefined;
     return response;
   }
+
+  private getBlobAndContainerNamesFromUrl(
+    url: string
+  ): { blobName: string; containerName: string } {
+    url = url.split("?")[0]; // removing the sas part of url if present
+    const urlWithoutBlobName = url.substring(0, url.lastIndexOf("/"));
+    return {
+      blobName: url.substring(url.lastIndexOf("/") + 1, url.length),
+      containerName: urlWithoutBlobName.substring(
+        urlWithoutBlobName.lastIndexOf("/") + 1,
+        urlWithoutBlobName.length
+      )
+    };
+  }
+
+  /**
+   * ONLY AVAILABLE IN NODE.JS RUNTIME.
+   *
+   * @memberof BlobClient
+   */
+  public generateSASUrl(
+    options: GenerateSASUrlOptions,
```

Comment:
[@jeremymeng's comment from HarshaNalluru/pull/21 ]

I believe we don't want to expose full set of SAS token. So should only take 'r' permission and this would be generatePublicReadOnlyUrl or similar.